opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreation.jsx

```diff
@@ -351,37 +352,44 @@ class StixCoreRelationshipCreation extends Component {
       defaultCreatedBy,
       defaultMarkingDefinitions,
     } = this.props;
-    const relationshipTypes = resolveRelationsTypes(
-      fromObjects[0].entity_type,
-      toObjects[0].entity_type,
-    );
     return (
-      <>
-        <div className={classes.header}>
-          <IconButton
-            aria-label="Close"
-            className={classes.closeButton}
-            onClick={this.handleClose.bind(this)}
-            size="large"
-          >
-            <Close fontSize="small" color="primary" />
-          </IconButton>
-          <Typography variant="h6">{t('Create a relationship')}</Typography>
-        </div>
-      <StixCoreRelationshipCreationForm
-        fromEntities={fromObjects}
-        toEntities={toObjects}
-        relationshipTypes={relationshipTypes}
-        handleReverseRelation={this.handleReverseRelation.bind(this)}
-        onSubmit={this.onSubmit.bind(this)}
-        handleClose={this.handleClose.bind(this)}
-        defaultConfidence={confidence}
-        defaultStartTime={startTime}
-        defaultStopTime={stopTime}
-        defaultCreatedBy={defaultCreatedBy}
-        defaultMarkingDefinitions={defaultMarkingDefinitions}
-      />
-      </>
+        <UserContext.Consumer>
+          {({ schema }) => {
+            const relationshipTypes = resolveRelationsTypes(
+              fromObjects[0].entity_type,
+              toObjects[0].entity_type,
+              schema.schemaRelationsTypesMapping,
+            );
+            return (
+                <>
+                  <div className={classes.header}>
+                    <IconButton
+                      aria-label="Close"
+                      className={classes.closeButton}
+                      onClick={this.handleClose.bind(this)}
+                      size="large"
+                    >
+                      <Close fontSize="small" color="primary" />
+                    </IconButton>
+                    <Typography variant="h6">{t('Create a relationship')}</Typography>
+                  </div>
+                  <StixCoreRelationshipCreationForm
+                      fromEntities={fromObjects}
+                      toEntities={toObjects}
+                      relationshipTypes={relationshipTypes}
+                      handleReverseRelation={this.handleReverseRelation.bind(this)}
+                      onSubmit={this.onSubmit.bind(this)}
+                      handleClose={this.handleClose.bind(this)}
+                      defaultConfidence={confidence}
+                      defaultStartTime={startTime}
+                      defaultStopTime={stopTime}
+                      defaultCreatedBy={defaultCreatedBy}
+                      defaultMarkingDefinitions={defaultMarkingDefinitions}
+                  />
+                </>
+            );
+          }}
+        </UserContext.Consumer>
```

Comment:
For curiosity, can you tell me what is `UserContext.Consumer` tag, and why embedding the rest of the code in it?

Reply:
This allows the application context to be used in React classes. It's the old way before hooks.

https://legacy.reactjs.org/docs/context.html